### ISSUE TITLE
Column identifiers not wrapped for rowNumber()

### ISSUE THREAD

**Issue by licitdev** [knex/knex#5345]
# Environment

Knex version: 2.2.0
Database + version: Oracle 18
OS: macOS

# Bug

The identifiers are not wrapped for analytic functions such as `rowNumber()`, leading to the column identifier being capitalised for Oracle. Other dialects worked as expected.

```sql
row_number() over (
    partition by "a"."id"
    order by
        "b"."name" desc
) as row_no
```
The `row_no` is unquoted, which becomes `ROW_NO` in the result set.

A workaround was used in  https://github.com/directus/directus/pull/14798/commits/a664d68109684fa28f935b36ca7099c928082170 by parsing the column identifier with `raw()`.

# Sample Code

```ts
queryBuilder.rowNumber("row_no", knex.raw(`partition by ?? order by`, ["a.id", "b.name"]));
```

**Comment by mercmobily** [comment]
You’re right: the alias for analytic functions like `rowNumber()` isn’t wrapped, which causes Oracle to fold it to upper‑case. This needs further work on the compiler side.

Analytic columns are compiled in the generic query compiler, and the alias is appended as a raw `as <alias>` string (lib/query/querycompiler.js:1180–1215 for `with()` right above, and the analytic emitter just below, which ends with `if (stmt.alias) sql += ' as ' + stmt.alias;`). In contrast, regular columns and aggregates go through `wrap()/columnize()` which quote identifiers for the current dialect.

Two options to address this:

- Wrap the alias for analytic expressions using the same identifier wrapper as columns; or
- Allow passing a Raw alias when exact quoting is required.

Workaround today: pass a Raw alias or quote it yourself, e.g. `qb.rowNumber(knex.raw('"row_no"'), …)`.

References: lib/query/querycompiler.js:1180–1215 (and the analytic path just below)
